src/proxy_lib/proxy_lib.c

```diff
@@ -207,41 +169,32 @@ void *realloc(void *ptr, size_t size) {
         return NULL;
     }
 
-    size_t leak_pool_contains_pointer = ba_leak_pool_contains_pointer(ptr);
-    if (leak_pool_contains_pointer) {
-        return ba_leak_realloc(ptr, size, leak_pool_contains_pointer);
-    }
-
-    if (Proxy_pool) {
+    if (!was_called_from_umfPool && Proxy_pool) {
         was_called_from_umfPool = 1;
         void *new_ptr = umfPoolRealloc(Proxy_pool, ptr, size);
         was_called_from_umfPool = 0;
         return new_ptr;
     }
 
-    assert(0);
-    return NULL;
+    return ba_alloc_global_realloc(ptr, size);
 }
 
 void free(void *ptr) {
     if (ptr == NULL) {
         return;
     }
 
-    if (ba_leak_free(ptr) == 0) {
-        return;
-    }
-
-    if (Proxy_pool) {
+    if (!was_called_from_umfPool && Proxy_pool) {
```

Comment:
**(Important)** What if an app wants to free a pointer from the base allocator when `Proxy_pool` already exists?
How can you distinguish which allocator a pointer comes from?